src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,207 @@
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NullTestAroundInstanceOf&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *
+ *     if (myObj != null &amp;&amp; myObj instanceof Object) { // violation
+ *
+ *     }
+ *
+ *     if (myObj != null) { // violation
+ *         if (myObj instanceof Object) {
+ *
+ *         }
+ *     }
+ *
+ *    if ((myObj != null) &amp;&amp; (myObj instanceof Object)) {}  // violation
+ * }
+ * </pre>
+ *
+ * @since 8.34
+ */
+@StatelessCheck
+public class NullTestAroundInstanceOfCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message null.before.instanceof in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "null.test.around.instanceof";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_IF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST landNode = ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(TokenTypes.LAND);
+        if (landNode == null) {
+            if (checkOnNextLine(ast)) {
+                log(ast, MSG_KEY);
+            }
+        }
+        else {
+            final DetailAST notEqualNode = landNode.findFirstToken(TokenTypes.NOT_EQUAL);
+            final DetailAST instanceOfNode =
+                    landNode.findFirstToken(TokenTypes.LITERAL_INSTANCEOF);
+            if (notEqualNode != null && instanceOfNode != null
+                    && notEqualNode.findFirstToken(TokenTypes.LITERAL_NULL) != null
+                    && getIdentText(notEqualNode).equals(getIdentText(instanceOfNode))) {
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * To get instanceOfNode or notEqualNode's full ident text.
+     *
+     * @param ast DetailAST
+     * @return string ident node's text
+     */
+    private static String getIdentText(DetailAST ast) {
+        final DetailAST dotNode = ast.findFirstToken(TokenTypes.DOT);
+        final String string;
+        if (dotNode == null) {
+            string = ast.findFirstToken(TokenTypes.IDENT).getText();
+        }
+        else {
+            final FullIdent fullIdent = FullIdent.createFullIdent(dotNode);
+            string = fullIdent.getText();
+        }
+        return string;
+    }
+
+    /**
+     * Check if there is a unnecessary 'null' check before 'instanceof' expression when
+     * 'instanceof' expression is on next line of 'null' check and vice versa.
+     *
+     * @param ast DetailAST
+     * @return error boolean
+     */
+    private static boolean checkOnNextLine(DetailAST ast) {
+        final DetailAST outerNotEqualNode = ast.findFirstToken(TokenTypes.EXPR)
```

Comment:
can be divide logic inside this method to two separate methods? one method will check for outer not equal node and other method will check for outer instance of node

Reply:
done